my-account/app/myaccount/login/LogInPageController.java

```diff
@@ -0,0 +1,165 @@
+import io.sphere.sdk.customers.commands.CustomerSignInCommand;
+import io.sphere.sdk.customers.errors.CustomerInvalidCredentials;
+import myaccount.CustomerSessionUtils;
+import play.Logger;
+import play.data.Form;
+import play.filters.csrf.AddCSRFToken;
+import play.filters.csrf.RequireCSRFCheck;
+import play.libs.F;
+import play.mvc.Result;
+import play.twirl.api.Html;
+import shoppingcart.CartSessionUtils;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import static myaccount.CustomerSessionUtils.overwriteCustomerSessionData;
+import static shoppingcart.CartSessionUtils.overwriteCartSessionData;
+
+/**
+ * Allows to log in as a customer.
+ */
+@Singleton
+public final class LogInPageController extends SunriseController {
+
+    private static final Form<LogInFormData> LOG_IN_FORM = Form.form(LogInFormData.class);
+    private static final Form<SignUpFormData> SIGN_UP_FORM = Form.form(SignUpFormData.class);
+
+    @Inject
+    public LogInPageController(final ControllerDependency controllerDependency) {
+        super(controllerDependency);
+    }
+
+    public F.Promise<Result> show(final String languageTag) {
+        final UserContext userContext = userContext(languageTag);
+        final LogInPageContent pageContent = new LogInPageContent();
+        return F.Promise.pure(ok(renderLogInPage(pageContent, userContext)));
+    }
+
+    @AddCSRFToken
+    @RequireCSRFCheck
+    public F.Promise<Result> processLogIn(final String languageTag) {
+        final UserContext userContext = userContext(languageTag);
+        final Form<LogInFormData> form = LOG_IN_FORM.bindFromRequest();
+        final LogInPageContent pageContent = new LogInPageContent(form);
+        if (form.hasErrors()) {
+            return F.Promise.pure(handleLogInFormErrors(form, pageContent, userContext));
+        } else {
+            return logIn(form)
+                    .map(signInResult -> handleSuccessfulSignIn(signInResult, userContext))
+                    .recover(throwable -> handleInvalidCredentialsError(throwable, pageContent, userContext));
+        }
+    }
+
+    @AddCSRFToken
+    @RequireCSRFCheck
+    public F.Promise<Result> processSignUp(final String languageTag) {
+        final UserContext userContext = userContext(languageTag);
+        final Form<SignUpFormData> form = SIGN_UP_FORM.bindFromRequest();
+        final LogInPageContent pageContent = new LogInPageContent(form, userContext, i18nResolver(), configuration());
+        if (form.hasErrors()) {
+            return F.Promise.pure(handleSignUpFormErrors(form, pageContent, userContext));
+        } else {
+            return signUp(form)
+                    .map(signInResult -> handleSuccessfulSignIn(signInResult, userContext))
+                    .recover(throwable -> handleExistingCustomerError(throwable, pageContent, userContext));
+        }
+    }
+
+    public Result processLogOut(final String languageTag) {
+        CustomerSessionUtils.removeCustomer(session());
```

Comment:
should we add here sth. more generic? What if another page part needs also to clean up the session?
Can we just clear the whole session?